### PR TITLE
Deprecate `DeviceResident.to_gpu`/`to_cpu`/`to_intel64`

### PR DESCRIPTION
**Pull request by niboshi**
Deprecates  `DeviceResident.to_gpu`/`to_cpu`/`to_intel64` in favor of `to_device`.

### CODE REVIEW

File: chainer/device_resident.py [hunk 1]
```diff
@@ -248,7 +262,22 @@ def __init__(
...
+            elif method == 'to_gpu':
+                message = (
+                    'to_gpu is deprecated. '
+                    'Please use to_device(\'@cupy:N\') instead.')
+            elif method == 'to_intel64':
+                message = (
+                    'to_intel64 is deprecated. '
+                    'Please use to_device(\'@intel64\') instead.')
+            else:
+                assert False
```
Comment (emcastillo) [inline chainer/device_resident.py]: Although the verbosity is not bad in this case, just to be pedantic code could be reduced a bit with:

```python
device_names = {'to_cpu': '@numpy', 'to_gpu': '@cupy:N', 'to_intel64':'@intel64'}
if method in device_names:
    message = '{} is deprecated. Please use to_device(\'{}\') instead.'.format(method, device_names[method])
else:
    assert False
```

There's no need to change it though.

Comment (niboshi) [reply chainer/device_resident.py]: Thank you for suggestion. Fixed in 1658faf8e0e72b11b6bbadd00502771cb487624b.

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Fixed test failures. PTAL

**Comment by emcastillo** [comment]
Jenkins, test this please

**Pull request merged**